web/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/api/JobAdvertisementFromApiAssembler.java

```
@@ -202,10 +200,28 @@ private PublicContactDto convertPublicContact(ApiPublicContactDto apiPublicConta
 				.setSalutation(apiPublicContact.getSalutation())
 				.setFirstName(trimOrNull(apiPublicContact.getFirstName()))
 				.setLastName(trimOrNull(apiPublicContact.getLastName()))
-				.setPhone(trimOrNull(apiPublicContact.getPhone()))
+				.setPhone(sanitizePhoneNumber(trimOrNull(apiPublicContact.getPhone())))
 				.setEmail(trimOrNull(apiPublicContact.getEmail()));
 	}
 
+	/*
+	 * Check for a valid phone number and format as international number (with spaces)
+	 * example: +41 79 555 12 34
+	 */
+	private String sanitizePhoneNumber(String phone) {
+		if (hasText(phone)) {
+			try {
+				Phonenumber.PhoneNumber phoneNumber = PhoneNumberUtil.getInstance().parse(phone, "CH");
+				if (PhoneNumberUtil.getInstance().isValidNumber(phoneNumber)) {
+					return PhoneNumberUtil.getInstance().format(phoneNumber, PhoneNumberUtil.PhoneNumberFormat.INTERNATIONAL);
+				}
+			} catch (NumberParseException e) {
+				LOGGER.warn("JobAdvertisementFromApiAssembler.convert ::: ApiCreateJobAdvertisementDto has invalid phone number: {}", phone);
```

I think we should convert the ``NumberParseException`` RuntimeException subclass and return an error to the client, instead of nulling field.

---

Here could be send back a validation exception

---

Ok, I'll add it.